### PR TITLE
add setTimeout example with let and var

### PR DESCRIPTION
**Pull request by superbuggy**
I wanted to deepen the coverage of the differences between function-scoping and block-scoping

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ ES6 introduces the `let` keyword which works just like `var` but is scoped to it
 > Code blocks should be used as parts of special forms like for function bodies or the bodies of the flow control forms listed above.
 
 Below are contrived examples of isolating a variable `a` to a local scope.
-Pre-es6 we need to use a verbose IIFE (immediately invoked function expression) to create an isolate scope.
+Prior to ES6, we needed to use something called an IIFE (immediately invoked function expression) to place the `a` (which is a `var`) an isolated scope. This is because `var`s are ***function-scoped***. In contrast `let` statements are ***block scoped***, meaning they don't exist or are not accessible outside of curly braces `{}`.
```
Comment (amaseda) [inline readme.md]: "an isolated scope" --> "in an isolated scope"

File: readme.md [hunk 2]
```diff
@@ -121,6 +121,30 @@ for(let j = 0; j < 10; j++){
 console.log("outside loop:", j)
 //throws an error
 ```
+
+The function-scoped of `var` and block-scoped behavior of `let` can have some initially puzzling but important and interesting differences in certain situations. We will delay the `console.log(i)` operation, or evaluating `i`, with `setTimeout` to highlight an important distinction in how **scope** with `let` and `var` operates differently.
```
Comment (amaseda) [inline readme.md:125]: "function-scoped of var" --> "function-scoped behavior"

Comment (amaseda) [reply readme.md:125]: "but important and interesting differences" --> "but important differences"

File: readme.md [hunk 3]
```diff
@@ -121,6 +121,30 @@ for(let j = 0; j < 10; j++){
...
+
+In the following scenario, since `var` is ***not block-scoped***, it "leaks" outside of the for-loop. This is not the case with `let` which is **block-scoped**.
```
Comment (amaseda) [inline readme.md]: "case with `let` which" --> "case with `let`, which"

**Comment by superbuggy** [comment]
I kept "important and interesting" (curiosity > anxiety framing)

**Pull request merged**